compiler/src/iree/compiler/Codegen/Common/TransformExtensions/CommonExtensions.cpp

```diff
@@ -924,31 +924,28 @@ void transform_dialect::WorkgroupSwizzleOp::getEffects(
 
 static void setAnchorOpsFromAttributes(VectorLayoutAnalysis &analysis,
                                        func::FuncOp funcOp) {
-  for (Block &block : funcOp) {
-    for (Operation &op : block) {
-      for (NamedAttribute attr : op.getAttrs()) {
-        StringRef name = attr.getName().strref();
-        if (name.find("__vector_layout_test_anchor_operand_") !=
-            std::string::npos) {
-          int operandNum;
-          name.substr(name.find_last_of("_") + 1)
-              .getAsInteger(/*Radix=*/10, operandNum);
-          assert(operandNum < op.getNumOperands() &&
-                 "operand number out of range");
-          analysis.setAnchor(op.getOperand(operandNum), attr.getValue());
-        }
-        if (name.find("__vector_layout_test_anchor_result_") !=
-            std::string::npos) {
-          int resultNum;
-          name.substr(name.find_last_of("_") + 1)
-              .getAsInteger(/*Radix=*/10, resultNum);
-          assert(resultNum < op.getNumResults() &&
-                 "result number out of range");
-          analysis.setAnchor(op.getResult(resultNum), attr.getValue());
-        }
+  funcOp.walk([&](Operation *op) {
+    for (NamedAttribute attr : op->getAttrs()) {
+      StringRef name = attr.getName().strref();
+      if (name.find("__vector_layout_test_anchor_operand_") !=
+          std::string::npos) {
+        int operandNum;
+        name.substr(name.find_last_of("_") + 1)
+            .getAsInteger(/*Radix=*/10, operandNum);
+        assert(operandNum < op->getNumOperands() &&
+               "operand number out of range");
```

Comment:
Maybe not in this PR, but we should do something better than substring checks lol

Reply:
Maybe, but this is purely for testing so I don't think it's an issue.